changelogs/unreleased/gh-6144-incorrect-timeout-error.md

```diff
@@ -0,0 +1,3 @@
+## bugfix/net.box
+
+* Fixed bug with the type of the error that occurs when timeout expires in net.box functions (gh-6144).
```

Comment:
- Line's too long (should be < 80 characters).
- I wouldn't call it a bug. I'd just say: `Changed the type of the error returned by net.box on timeout from ClientError to TimedOut (gh-6144).`
- Re commit message: `Closes #XXXX` should go before the doc bot request (otherwise it will be appended to the request).
- Re doc bot request: I looked through the [net.box documentation](https://www.tarantool.io/en/doc/latest/reference/reference_lua/net_box/) and didn't see any mention of what kind of error is returned on timeout. It looks like we can drop the doc bot request, as there's nothing to fix in the doc.

Reply:
Fixed